src/lib.rs

```diff
@@ -110,53 +183,61 @@ fn random_gen_dust_concentration() -> String {
     rng.gen_range(0..=1000).to_string()
 }
 
-pub fn run(cli: &Cli) {
-    println!("\r\ncli: {:?}\r\n", cli);
-
+fn send_request(url: &str, client: Client) -> reqwest::Result<Response> {
     let dust_concentration = random_gen_dust_concentration();
     let air_purity = AirPurity::from_value(dust_concentration.parse::<u16>().unwrap()).to_string();
     let reading = Reading::new(
         random_gen_temperature(),
         random_gen_humidity(),
         random_gen_pressure(),
         dust_concentration,
-        air_purity
+        air_purity,
     );
 
     let json = serde_json::to_string(&reading).unwrap();
-    const URL: &str = "http://localhost:4000/api/readings/add";
-
-    println!("Sending POST request to {} as JSON: {}", URL, json);
-
-    let client = Client::new();
-    let res = client
-        .post(URL)
+    println!("Sending POST request to {} as JSON: {}", url, json);
+    client
+        .post(url)
         .header(CONTENT_TYPE, "application/json")
         .body(json)
-        .send();
-    match res {
-        Ok(response) => {
-            match cli.debug {
-                true => println!("Response from Ambi backend: {:#?}", response),
-                false => println!("Response from Ambi backend: {:?}", response.status().as_str())
-            }
-        }
-        Err(e) => {
-            match cli.debug {
-                // Print out the entire reqwest::Error for verbose debugging
-                true => eprintln!("Response error from Ambi backend: {:?}", e),
-                // Keep the error reports more succinct
-                false => {
-                    if e.is_request() {
-                        eprintln!("Response error from Ambi backend: request error");
-                    } else if e.is_timeout() {
-                        eprintln!("Response error from Ambi backend: request timed out");
-                    } else {
-                        eprintln!("Response error from Ambi backend: specific error type unknown");
-                    }
-                }
+        .send()
+}
+
+pub fn run(cli: &Cli) {
+    println!("\r\ncli: {:?}\r\n", cli);
+
+    const URL: &str = "http://localhost:4000/api/readings/add";
+    let mut handlers: Vec<JoinHandle<()>> = vec![];
+    let debug: bool = cli.debug;
+    for _ in 0..cli.int {
+        let handler = spawn(move ||
+            match send_request(URL, Client::new()) {
+              Ok(response) => {
+                Output::new(
```

Comment:
In reference to my above suggestion for a different name and a static version, this then might look something like:

```
ResponseLog.print(
    None,
    Some(response),
    std::thread::current().id()
);
```

The `String` could become implicit because it's a specific enough kind of logger type, and you could initialize the static instance of it with `debug` once instead of needing to pass it along every time.

One other thought: could the None and Some become a literal `Option` that you pass to `print()` in which it could figure out whether it's a success message vs an error message?

Just an idea, curious what you think.